Memory pair game

webdevtoday opened the pull request:
### memory-pair-game

[Demo](https://webdevtoday.github.io/memory-pair-game/) |
[Code base](https://github.com/webdevtoday/memory-pair-game)

The code is submitted in a dedicated feature branch.
Please, review.

MaxKalashnyk commented on submissions/webdevtoday/memory-pair-game/js/app.js:
```diff
@@ -0,0 +1,115 @@
+    }
+
+    function checkWin() {
+        if (numberOfOpenCards === 12) {
```
Move `12` to the constants

webdevtoday replied:
Yes, it's my favorite mistake))

MaxKalashnyk commented on submissions/webdevtoday/memory-pair-game/js/app.js:
```diff
@@ -0,0 +1,115 @@
+            showScore();
+            checkWin();
+            blockClicks = false;
+        }, 500);
```
Move `500` time value to the constants

webdevtoday replied:
I'll fix it

MaxKalashnyk reviewed (changes requested):
Nice work! I'd recommend you adding some button to start a new game after finishing playing.

webdevtoday commented on submissions/webdevtoday/memory-pair-game/css/style.css:
```diff
@@ -0,0 +1,96 @@
+    margin: 0 auto;
+}
+
+.flip-container {
```
That's true, I forgot it, I did it, generally, on my phone))) I'll fix it!

webdevtoday commented on submissions/webdevtoday/memory-pair-game/js/app.js:
```diff
@@ -0,0 +1,115 @@
+    let blockClicks = false;
+
+    function rand(min, max) {
+        let rand = min + Math.random() * (max + 1 - min);
```
You're right! I'll fix it

webdevtoday commented on submissions/webdevtoday/memory-pair-game/js/app.js:
```diff
@@ -0,0 +1,115 @@
+                </div>
+            </div>
+        `;
+        const card = cardTemplate.replace('{{imgpath}}', imgPathBuilder( randomCutImage() ));
```
Ohh, I didn't think about it, I'll change it

webdevtoday commented:
> Nice work! I'd recommend you adding some button to start a new game after finishing playing. Look at my comments below as well

Thank you very much! I'll fix all of the problems.

webdevtoday commented:
@MaxKalashnyk Hi! I fixed problems and I added New Game button.

MaxKalashnyk commented on submissions/webdevtoday/oop_exercise/index.js:
```diff
@@ -1,3 +1,75 @@
+<<<<<<< HEAD
```
I think you don't need this file in this PR

webdevtoday replied:
I've lost a part of work when I made commits( and I got this file. I don't understand how to fix it(

MaxKalashnyk reviewed (changes requested):
I don't see that all of the issues are resolved. Please, fix all of them and we'll ready to approve this PR

webdevtoday commented:
@MaxKalashnyk Hi! I tried to fix my problems. I revert my last commit and then I made changes in my code. Then I commit it and made steps from instruction to commit.

MaxKalashnyk commented:
Your new commit affected all of the student's works in the repository. Please, fix it

webdevtoday commented:
> Your new commits affected all of the student's works in the repository. Please, fix it

I don't know how to do it, could you help me, please?

MaxKalashnyk commented:
Go to your `webdevtoday:memory-pair-game` branch. 
Then you can run command `git rebase -i main`. After running this command you can type `i` key in order to have an ability to edit all of your commits. Choose your last commits that broke your PR and type the `d` flag near them. It means that you'll drop them from your PR. After doing it, press `i` again. And run `:wq` for saving this configuration.
More info about the `git rebase` command you can find [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase).

Let me know if you need any help

webdevtoday commented:
> Go to your `webdevtoday:memory-pair-game` branch. 
> Then you can run command `git rebase -i main`. After running this command you can type `i` key in order to have an ability to edit all of your commits. Choose your last commits that broke your PR and type the `d` flag near them. It means that you'll drop them from your PR. After doing it, press `i` again. And run `:wq` for saving this configuration.
> More info about the `git rebase` command you can find [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase).
> 
> Let me know if you need any help

Ok, thank you very much! I'll try to do it

webdevtoday commented:
@MaxKalashnyk Hi! Please, check my PR, I've just done those steps.

webdevtoday commented:
@MaxKalashnyk I deleted all of my commits. And then I added my files again to project fold
![Снимок](https://user-images.githubusercontent.com/39851525/102580630-97a9db80-4107-11eb-9e9b-83283f8b4858.PNG)
![Снимок1](https://user-images.githubusercontent.com/39851525/102580632-98427200-4107-11eb-8522-23401be181f6.PNG)
![Снимок2](https://user-images.githubusercontent.com/39851525/102580635-98427200-4107-11eb-876b-772235d75a83.PNG)
![Снимок3](https://user-images.githubusercontent.com/39851525/102580637-98db0880-4107-11eb-9dfa-60875b249df8.PNG)



er.

webdevtoday commented:
@MaxKalashnyk But I had a problem at the end. I added screens of my console with my actions. I don't understand, how it happened with me, I did PR-s to homeworks repo many times.

MaxKalashnyk commented:
> @MaxKalashnyk But I had a problem at the end. I added screens of my console with my actions. I don't understand, how it happened with me, I did PR-s to homeworks repo many times.

I saw your screenshots. You ran all commands properly till the `git push --set-upstream origin memory-pair-game`.
As you have already created this remote upstream before, you didn't need to create it one more time.
I think in that case using `git push origin memory-pair-game` or just `git push` would help.

MaxKalashnyk reviewed (approved):
LGTM!
I see you managed to fixed your irrelevant commits.
I think we can approve this work. I hope you can do more complex and intersting designs of your apps in future. Now it's quite simple :) 
But it's no big deal. You did a great job!

webdevtoday commented:
@MaxKalashnyk Thank you! Thank you!Thank you!!!! Thank you very much! You're really help me!!
